### PR TITLE
provide Windows-compatible commands for make targets

### PR DESCRIPTION
**Pull request by bacongobbler**
Signed-off-by: Matthew Fisher <matt.fisher@microsoft.com>

### CODE REVIEW

**Comment by bacongobbler** [comment]
The only place I haven't found a clean equivalent for is `find -type d -exec` which is needed for `make dist`. Leaving that alone for the time being.

File: Makefile [hunk 1]
```diff
@@ -1,28 +1,40 @@
...
+GIT_TAG     = $(shell git describe --tags --abbrev=0 --exact-match 2>$(NUL))
+GIT_DIRTY   = $(shell test -n "`git status --porcelain`" && echo "dirty" || echo "clean")
+
+BINDIR      := $(CURDIR)/bin
+DIST_DIRS   := find * -type d -exec
+TARGETS     := darwin/amd64 linux/amd64 linux/386 linux/arm linux/arm64 linux/ppc64le windows/amd64
+
+ifeq ($(OS),Windows_NT)
+	BINNAME    = $(PROJECTNAME).exe
+	SHELL      = cmd.exe
```
Comment (mattfarina) [inline Makefile]: What is the `SHELL` environment variable used for? I see the `shell` make function is used to shell out. Am I missing something?

**Review by mattfarina — COMMENTED** [review]
I wonder if you can combine the `dir` and `call` commands on Windows in place of `find` for *nix.

File: Makefile (hunk 1 above)
Comment (bacongobbler) [reply Makefile]: From https://www.gnu.org/software/make/manual/html_node/Choosing-the-Shell.html:

> The program used as the shell is taken from the variable SHELL. If this variable is not set in your makefile, the program /bin/sh is used as the shell.

By setting this to cmd.exe (and other windows-equivalent variables), the Makefile will work without requiring cygwin or equivalent shell emulators installed.

**Comment by bacongobbler** [comment]
> I wonder if you can combine the `dir` and `call` commands on Windows in place of `find` for *nix.

I'd love to see an example! I tried looking for something similar but could not find a solution.

File: Makefile (hunk 1 above)
Comment (mattfarina) [reply Makefile]: Prior to setting the `SHELL` to `cmd.exe` on Windows there a variables set that use the `shell` function. For example,

```
GIT_COMMIT  = $(shell git rev-parse HEAD)
```

Does `SHELL` being set to `cmd.exe` need to happen earlier in the script?

**Comment by mattfarina** [comment]
@bacongobbler found a windows command set that may work...

```
for /F "tokens=*" %i in ('dir /d /b') do echo %i
```

This will loop over the output of `dir /d /b` and pipe the output to the command, in this case `echo`. The `/d` switch on dir only lists directories and the `/b` switch makes it a bare list.

**Comment by bacongobbler** [comment]
I'll come back and revisit this another time. WSL2 has been working great for me so far on Windows so the need for a windows-compatible makefile has been less important now.

**Pull request closed without merging**